modules/h2/src/main/java/org/gridgain/internal/h2/table/Table.java

```diff
@@ -744,7 +744,7 @@ public PlanItem getBestPlanItem(Session session, int[] masks,
             AllColumnsForPlan allColumnsSet, boolean isEquiJoined) {
         PlanItem item = new PlanItem();
         item.setIndex(getScanIndex(session));
-        item.cost = item.getIndex().getCost(session, null, filters, filter, null, allColumnsSet);
+        item.cost = item.getIndex().getCost(session, null, filters, filter, sortOrder, allColumnsSet);
```

Comment:
I'll revert it to not change H2, but I believe that it should be work better with it.

Reply:
Actually I didn't mean that it should be reverted =) My point was that we should know about all pitfalls in case this particular change will be omitted